### PR TITLE
DSEGOG-53 right y axis

### PR DESCRIPTION
**Pull request by louise-davies**
## Description
Allows users to switch channels between the left and right Y axis. The right Y axis is only visible if at least one channel is assigned to it.

Here's an screenshot showing the second Y axis in action, as well as showing the menu control

image

## Testing instructions
Add a set up instructions describing how the reviewer should test the code

- [ ] Review code
- [ ] Check Actions build
- [ ] Review changes to test coverage
- [ ] {more steps here}

## Agile board tracking
Closes DSEGOG-53

### CODE REVIEW

**Review by sam-glendenning — CHANGES_REQUESTED** [review]
Code looks good so far but see the below:

**Comment by sam-glendenning** [comment]
Oh whoops, that didn't work. Let me try again

File: src/plotting/plot.component.tsx [hunk 1]
```diff
@@ -121,7 +121,9 @@ const Plot = (props: PlotProps) => {
         },
         y2: {
           type: YAxesScale,
-          display: false,
+          display: selectedPlotChannels.some(
+            (channel) => channel.options.yAxis === 'right'
+          ),
```
Comment (sam-glendenning) [inline src/plotting/plot.component.tsx:126]: We have this check for the right y-axis but could we also add the equivalent for the left one? If a user wants to, they could do this:

image

which looks a little confusing

**Review by sam-glendenning — CHANGES_REQUESTED** [review]
Here we go:

File: src/plotting/plotSettings/moreOptions/moreOptionsBox.component.tsx [hunk 2]
```diff
@@ -143,6 +169,53 @@ const MoreOptionsBox = (props: MoreOptionsProps) => {
...
+                componentsProps={{ typography: { sx: { fontSize: 12 } } }}
+                sx={{ margin: 0 }}
+              />
+              <FormControlLabel
+                value="right"
+                control={<Radio size="small" sx={{ padding: '2px' }} />}
+                label="Right"
+                componentsProps={{ typography: { sx: { fontSize: 12 } } }}
+                sx={{ margin: 0 }}
+              />
```
Comment (sam-glendenning) [inline src/plotting/plotSettings/moreOptions/moreOptionsBox.component.tsx:215]: Could we add aria labels for these?

File: src/plotting/plot.component.tsx (hunk 1 above)
Comment (louise-davies) [reply src/plotting/plot.component.tsx:126]: We should check this with CLF - there's a specific bit in the requirements to hide the right Y axis if nothing is assigned to it but nothing for the left one

File: src/plotting/plotSettings/moreOptions/moreOptionsBox.component.tsx (hunk 2 above)
Comment (louise-davies) [reply src/plotting/plotSettings/moreOptions/moreOptionsBox.component.tsx:215]: These should all be accessible due to using labelled-by and the fact that radios are labelled by text

**Comment by louise-davies** [comment]
@sam-glendenning should have probably only tagged you for review after I had put this out of draft - I haven't written any tests or anything 😆

File: src/plotting/plot.component.tsx (hunk 1 above)
Comment (sam-glendenning) [reply src/plotting/plot.component.tsx:126]: If we don't hide the left one, some values have to be assigned to it, which Chart.js does. And it potentially results in some confusion as it looks like one channel maps to two y-values. That was my reasoning

Comment (louise-davies) [reply src/plotting/plot.component.tsx:126]: True, should probably just ensure that this case never occurs

File: src/plotting/plotSettings/moreOptions/moreOptionsBox.component.tsx (hunk 2 above)
Comment (sam-glendenning) [reply src/plotting/plotSettings/moreOptions/moreOptionsBox.component.tsx:215]: If you're happy, I'm happy. Didn't fancy busting out the screen reader for it ;)

File: src/plotting/plot.component.tsx (hunk 1 above)
Comment (sam-glendenning) [reply src/plotting/plot.component.tsx:126]: Either that or give the user the freedom to plot on the right y-axis while hiding the left one (I know this isn't usually done in practice but in case they wanted to)

File: src/plotting/plotSettings/moreOptions/moreOptionsBox.component.tsx (hunk 2 above)
Comment (louise-davies) [reply src/plotting/plotSettings/moreOptions/moreOptionsBox.component.tsx:215]: Inspecting accessibility properties on FF shows that all the roles and labels look fine & associated correctly

image

File: src/plotting/plotSettings/moreOptions/moreOptionsBox.component.test.tsx [hunk 3]
```diff
@@ -122,4 +128,27 @@ describe('MoreOptionsBox', () => {
...
+        lineStyle: 'solid',
+        yAxis: i === 1 ? 'right' : 'left',
+      },
+    }));
+
+    createView();
+
+    await user.click(
+      screen.getByLabelText(`Pick ${props.channel.name} colour`)
+    );
```
Comment (sam-glendenning) [inline src/plotting/plotSettings/moreOptions/moreOptionsBox.component.test.tsx]: Old copy-pasted code?

Comment (louise-davies) [reply src/plotting/plotSettings/moreOptions/moreOptionsBox.component.test.tsx]: Oops - good spot! Yeah I just copied the previous test as the base of this one 😅

**Review by sam-glendenning — APPROVED** [review]
